stories/__tests__/SQFormButton.stories.test.js

```diff
@@ -0,0 +1,161 @@
+import React from 'react';
+import {render, screen} from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+import {composeStories} from '@storybook/testing-react';
+import * as stories from '../SQFormButton.stories';
+
+const {
+  Default: SQFormButton,
+  WithTestField: SQFormButtonWithField
+} = composeStories(stories);
+
+describe('Default Tests', () => {
+  it('should render a button with text', () => {
+    render(<SQFormButton />);
+
+    const submitButton = screen.getByRole('button', {name: /form submission/i});
+
+    expect(submitButton).toBeInTheDocument();
+    expect(submitButton).toHaveTextContent('Submit');
+  });
+
+  it('should render a reset button given the type', () => {
+    render(<SQFormButton type="reset" />);
+
+    const resetButton = screen.getByRole('button', {name: /form submission/i});
+
+    expect(resetButton).toBeInTheDocument();
+    expect(resetButton).toHaveAttribute('type', 'reset');
+  });
+
+  it('should call a function when clicked', () => {
+    const onClickSpy = jest.fn();
+    render(<SQFormButton onClick={onClickSpy} />);
+
+    const submitButton = screen.getByRole('button', {name: /form submission/i});
+
+    userEvent.click(submitButton);
+
+    expect(onClickSpy).toHaveBeenCalled();
+  });
+
+  it('should show as disabled', () => {
+    render(<SQFormButton isDisabled={true} />);
+
+    const submitButton = screen.getByRole('button', {name: /form submission/i});
+
+    expect(submitButton).toBeDisabled();
+  });
+
+  it('should not fire onClick when disabled', () => {
+    const onClickSpy = jest.fn();
+    render(<SQFormButton isDisabled={true} onClick={onClickSpy} />);
+
+    const submitButton = screen.getByRole('button', {name: /form submission/i});
+
+    userEvent.click(submitButton);
+
+    expect(onClickSpy).not.toHaveBeenCalled();
+  });
+});
+
+describe('Button with Test Field Tests', () => {
+  it('should render a button and a text field', () => {
+    render(<SQFormButtonWithField />);
+
+    const testField = screen.getByLabelText(/test field/i);
+
+    expect(testField).toBeInTheDocument();
+
+    const submitButton = screen.getByRole('button', {name: /form submission/i});
+
+    expect(submitButton).toBeInTheDocument();
+    expect(submitButton).toHaveTextContent('Submit');
+  });
+
+  it('should disable when shouldRequireFieldUpdates is true', () => {
```

Comment:
nitpick: I think this test would fit better categorically in the default tests suite because it's testing a behavior that would happen regardless of if it was associated with an input field